main.go

```diff
@@ -101,21 +129,37 @@ func petasos(arguments []string) int {
 
 	infoLog.Log("configurationFile", v.ConfigFileUsed())
 
-	var (
-		accessor = new(service.UpdatableAccessor)
+	tracing, err := loadTracing(v, applicationName)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to build tracing component: %v \n", err)
+		return 1
+	}
+	infoLog.Log(logging.MessageKey(), "tracing status", "enabled", tracing.Enabled)
 
-		redirectHandler = &servicehttp.RedirectHandler{
-			KeyFunc:      device.IDHashParser,
-			Accessor:     accessor,
-			RedirectCode: http.StatusTemporaryRedirect,
-		}
+	rootRouter := mux.NewRouter()
+	otelMuxOptions := []otelmux.Option{
+		otelmux.WithPropagators(tracing.Propagator),
+		otelmux.WithTracerProvider(tracing.TracerProvider),
+	}
+	rootRouter.Use(otelmux.Middleware("mainSpan", otelMuxOptions...), candlelight.EchoFirstTraceNodeInfo(tracing.Propagator))
 
-		requestFunc      = logginghttp.SetLogger(logger, logginghttp.Header("X-Webpa-Device-Name", "device_id"), logginghttp.Header("Authorization", "authorization"))
-		decoratedHandler = alice.New(xcontext.Populate(requestFunc)).Then(redirectHandler)
+	accessor := new(service.UpdatableAccessor)
 
-		_, petasosServer, done = webPA.Prepare(logger, nil, metricsRegistry, decoratedHandler)
-		signals                = make(chan os.Signal, 10)
-	)
+	redirectHandler := &servicehttp.RedirectHandler{
+		KeyFunc:      device.IDHashParser,
+		Accessor:     accessor,
+		RedirectCode: http.StatusTemporaryRedirect,
+	}
+
+	requestFunc := logginghttp.SetLogger(logger, logginghttp.Header("X-Webpa-Device-Name", "device_id"), logginghttp.Header("Authorization", "authorization"), candlelight.InjectTraceInfoInLogger())
+	decoratedHandler := alice.New(xcontext.Populate(requestFunc)).Then(redirectHandler)
+
+	rootRouter.Handle("/{*}", decoratedHandler)
```

Comment:
We might not need to define all these handle calls if we use the http server instrumentation instead of the gorilla mux instrumentation. I'll see if I can hash a quick sample code for it. Here's an example: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/github.com/gorilla/mux/otelmux/v0.19.0/instrumentation/net/http/otelhttp/example/server/server.go#L71

Reply:
FYI: I used the gorilla mux instrumentation on the other services since they were already using it.

Reply:
I believe something like this would be more readable and still work as expected https://github.com/xmidt-org/petasos/compare/example/instrumentRedirector?expand=1

Reply:
utsavbatra5, let me know if you have any questions about this^

Reply:
No Joe. Sounds Good. Will release a new commit soon.